irohad/network/ordering_gate_common.hpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_ORDERING_GATE_COMMON_HPP
+#define IROHA_ORDERING_GATE_COMMON_HPP
+
+#include <memory>
+
+#include <boost/optional.hpp>
+#include "consensus/round.hpp"
+
+namespace shared_model {
+  namespace interface {
+    class Proposal;
+  }  // namespace interface
+}  // namespace shared_model
+
+namespace iroha {
+  namespace network {
+
+    /**
+     * Event, which is emitted by ordering gate, when it requests a proposal
+     */
+    struct OrderingEvent {
+      boost::optional<std::shared_ptr<shared_model::interface::Proposal>>
```

Comment:
Quite inefficient wrapping, we might consider custom overload similar to rust:
![image](https://user-images.githubusercontent.com/5658339/48712958-2c3b6a00-ec20-11e8-92db-a5743ec4242b.png)